### PR TITLE
Follow fog-core best practices

### PR DESCRIPTION
**Pull request by kddnewton**
Following https://github.com/fog/fog-core/pull/229, this gem needs to be namespaced to avoid deprecation warnings. This commit moves the files around to fit the better namespacing techniques. It additionally removes the `.ruby-version` file (so as to allow more easily testing and developing other Ruby versions) and adds Ruby 2.5.1 to the travis test suite.

### CODE REVIEW

**Comment by kddnewton** [comment]
cc @plribeiro3000 as you seem to be merging things the most to this gem these days

File: lib/fog/local.rb [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
 require 'fileutils'
 require 'tempfile'
-require File.expand_path('../local/version', __FILE__)
 
-module Fog
-  module Storage
-    autoload :Local, File.expand_path('../storage/local', __FILE__)
-  end
+require 'fog/local/version'
+require 'fog/local/storage'
```
Comment (plribeiro3000) [inline lib/fog/local.rb]: Prefer the `autoload` version please. =D

**Review by plribeiro3000 — COMMENTED** [review]
Just one small comment. Other then that LGTM.

File: lib/fog/local.rb (hunk 1 above)
Comment (kddnewton) [reply lib/fog/local.rb]: We can do that, but we just end up loading it immediately a couple of lines later. If that's preferable that's fine.

**Comment by plribeiro3000** [comment]
Cool! LGTM!

**Comment by geemus** [comment]
@kddeisz sorry for the temporary noise and thanks for the fix!

@plribeiro3000 thanks for review.

**Comment by plribeiro3000** [comment]
Released as `0.6.0`

**Pull request merged**